feat: README updated (#26)
goverland-labs/goverland-datasource-snapshot: README.md

![unit-tests](https://github.com/goverland-labs/goverland-datasource-snapshot/workflows/unit-tests/badge.svg)
![golangci-lint](https://github.com/goverland-labs/goverland-datasource-snapshot/workflows/golangci-lint/badge.svg)

Goverland Datasource Snapshot is a micro service designed to manage and synchronize various data sources for Goverland ecosystem.
At the moment the focus is on Snapshot API.

> The microservice also leverages technologies such as gRPC and NATS.

Reviewer: I think these details should be moved to other section (1-2 lines are base description, another lines are details about technoligies which were used)

Author: any ideas where?

> PostgresQL is used as database.

Reviewer: ```suggestion
PostgreSQL is used as database.
```

The application exposes Prometheus metrics.

## Usage

- The application initializes various services and workers to manage proposals, spaces, votes, and messages.

> - It sets up gRPC servers for remote procedure calls and integrates with NATS for event publishing.

Reviewer: ```suggestion
- Also it implements interface for voting on Snapshot  via gRPC calls.
```

## Features

- Snapshot SDK Integration: facilitates interaction with Snapshot services for proposals, spaces, votes, and messages.
- Event Publishing: utilizes NATS for event-driven architecture.

> - gRPC Services: provides gRPC-based APIs for interacting with the application.

Reviewer: ```suggestion
- gRPC Services: provides gRPC-based APIs for interacting with the Snapshot from internal services.
```

- Database: Integrates with PostgreSQL for storing and managing data.
- Metrics and Monitoring: exposes Prometheus metrics for monitoring.
- Health Checks: Implements health check endpoints to monitor the application's status.

## For Developers

### Pre-requisites

- GoLang 1.21
- Docker

### Run Locally

1. Clone the repository:
   ```bash
   git clone https://github.com/goverland-labs/goverland-datasource-snapshot.git
   cd goverland-datasource-snapshot
   ```
2. Install the required Go dependencies:
   ```bash

>     go mod tidy

Reviewer: ```suggestion
    go mod download
    go mod verify
```

   ```
3. Spin up the necessary containers:
   ```bash
   docker-compose up -d
   ```

> 4. Run the application. Don't forget to provide your own Snapshot API key:

Reviewer: Also we should describe the .env.dist file (it's a config for running the app)

Author: When I run the app locally via `go run main.go`, it looks like this config is ignored. What is the right way to do it?
Btw that's why I used the following command:
   ```bash
    POSTGRES_DSN="host=localhost port=5432 user=postgres password=postgres dbname=postgres sslmode=disable" \
    SNAPSHOT_API_KEY=YOUR_KEY \
    go run main.go
   ```

Reviewer: Moved to different PR

   ```bash
    POSTGRES_DSN="host=localhost port=5432 user=postgres password=postgres dbname=postgres sslmode=disable" \
    SNAPSHOT_API_KEY=YOUR_KEY \
    go run main.go
   ```

#### To tear down docker compose environment

```bash
docker-compose down
```

### Prometheus

Prometheus metrics are exposed at [http://localhost:2112/metrics](http://localhost:2112/metrics).

## Docker Image

1. Build the Docker Image:

   - Navigate to the root directory containing a `Dockerfile`.
   - Run the following command to build the Docker image:
     ```bash
     docker build -t goverland-datasource-snapshot .
     ```

2. Run the Docker Container:

   - After the image is built, you can run it as a container. Use the following command to start the container:
     ```bash

>      docker run -d --name goverland-datasource-snapshot -p 8080:8080 goverland-datasource-snapshot

Reviewer: ```suggestion
     docker run -d --name goverland-datasource-snapshot -p 11000:11000 goverland-datasource-snapshot
```
